File: ChatDemo-UI3.0/ChatDemo-UI3.0/Code/Controllers/Contacts/Cell/EMContactCell.m
/************************************************************
 *  * Hyphenate
 * __________________
 * Copyright (C) 2016 Hyphenate Inc. All rights reserved.
 *
 * NOTICE: All information contained herein is, and remains
 * the property of Hyphenate Inc.
 */

#import "EMContactCell.h"
#import "EMUserModel.h"

@interface EMContactCell()
@property (strong, nonatomic) IBOutlet UIImageView *avatarImage;
@property (strong, nonatomic) IBOutlet UILabel *nicknameLabel;

@end

@implementation EMContactCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
    self.accessoryType = UITableViewCellAccessoryNone;
}

- (void)setModel:(EMUserModel *)model {
    if (_model != model) {
        _model = model;
    }
    _nicknameLabel.text = _model.nickname;
    _avatarImage.image = _model.defaultAvatarImage;
    if (_model.avatarURLPath.length > 0) {
        [_avatarImage sd_setImageWithURL:[NSURL URLWithString:_model.avatarURLPath] placeholderImage:_model.defaultAvatarImage];
    }
    else {
        _avatarImage.image = _model.defaultAvatarImage;
    }
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

@end

File: swift-demo/Hyphenate-Demo-Swift/Code/Controllers/Contacts/Cell/EMContactCell.swift
//
//  EMContactCell.swift
//  Hyphenate-Demo-Swift
//
//  Created by 杜洁鹏 on 2017/6/26.
//  Copyright © 2017年 杜洁鹏. All rights reserved.
//

import UIKit
import SDWebImage

class EMContactCell: UITableViewCell {

    var _model: EMUserModel?
    
    @IBOutlet weak var avatarImage: UIImageView!
    @IBOutlet weak var nicknameLabel: UILabel!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        accessoryType = UITableViewCellAccessoryType.none
    }

    func set(model: EMUserModel) {
        if _model != model {
            _model = model
        }
        
        nicknameLabel.text = _model?.nickname
        avatarImage.image = _model?.defaultAvatarImage
        if _model?.avatarURLPath != nil {
            avatarImage.sd_setImage(with: URL(string: (_model?.avatarURLPath)!), placeholderImage: _model?.defaultAvatarImage)
        }
    }
    
    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
    }
    
}
